src/virt.c

```diff
@@ -715,9 +758,18 @@ refresh_lists (void)
             xpath_ctx = xmlXPathNewContext (xml_doc);
 
             /* Block devices. */
+	    char * bd_xmlpath=NULL;
+	    if( source == blockdevice_format ){
+		    bd_xmlpath="/domain/devices/disk/source[@dev]";
+	    }else{
+			if( dev == blockdevice_format ){
```

Comment:
Since I dont like this exact style of format, I changed it to smth slightly different than suggested, but I think logically it fixes the described issue.